lib/url.js

```diff
@@ -167,6 +168,20 @@ function isIpv6Hostname(hostname) {
   );
 }
 
+// This prevents some common spoofing bugs due to our use of IDNA toASCII. For
+// compatibility, the set of forbidden characters is the _intersection_ of
+// "forbidden host code point" in the WHATWG URL Standard and the characters in
+// the host parsing loop in Url.prototype.parse, with the following additions:
+//
+// - ':' since this could cause a "protocol spoofing" bug
+// - '@' since this could cause parts of the hostname to be confused with auth
+// - '[' and ']' since this could cause a non-IPv6 hostname to be interpreted
+//   as IPv6 by isIpv6Hostname above
+//
+// All four of these characters are also included in "forbidden host code
+// point". See https://url.spec.whatwg.org/#forbidden-host-code-point
```

Comment:
suggestion: Without reading the spec or any other materials, I'm confused by the way this comment says the four characters are included in "forbidden host code point" but also (above) that they are additions to the "forbidden host code point". I'm not sure my confusion here matters, but if there's an easy/obvious way to reword to clarify, let's do that.

Reply:
Removed this sentence and avoided referring to the set of characters we use as "forbidden host".